unbalanced_dataset/under_sampling/instance_hardness_threshold.py

```diff
@@ -0,0 +1,240 @@
+
+        threshold : float, optional (default=0.3)
+            Threshold to be used when excluding samples (0.01 to 0.99).
+
+        mode: str, optional (default='maj')
+            - If 'maj', only samples of the majority class are excluded.
+            - If 'all', samples of all classes are excluded.
+
+        cv : int, optional (default=5)
+            Number of folds to be used when estimating samples' instance hardness.
+
+        return_indices : bool, optional (default=False)
+            Either to return or not the indices which will be selected from
+            the majority class.
+
+        random_state : int or None, optional (default=None)
+            Seed for random number generation.
+
+        verbose : bool, optional (default=True)
+            Boolean to either or not print information about the processing
+
+        n_jobs : int, optional (default=-1)
+            The number of thread to open when it is possible.
+
+        Returns
+        -------
+        None
+
+        """
+        super(InstanceHardnessThreshold, self).__init__(
+            return_indices=return_indices,
+            random_state=random_state,
+            verbose=verbose)
+
+        self.estimator = estimator
+        self.threshold = threshold
+
+        possible_modes = ('maj', 'all')
+        if mode  not in possible_modes:
+            raise ValueError('Unknown mode parameter.')
+        else:
+            self.mode = mode
+
+        self.cv = cv
+
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """Find the classes statistics before to perform sampling.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            Matrix containing the data which have to be sampled.
+
+        y : ndarray, shape (n_samples, )
+            Corresponding label for each sample in X.
+
+        Returns
+        -------
+        self : object,
+            Return self.
+
+        """
+        # Check the consistency of X and y
+        X, y = check_X_y(X, y)
+
+        super(InstanceHardnessThreshold, self).fit(X, y)
+
+        return self
+
+    def transform(self, X, y):
+        """Resample the dataset.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            Matrix containing the data which have to be sampled.
+
+        y : ndarray, shape (n_samples, )
+            Corresponding label for each sample in X.
+
+        Returns
+        -------
+        X_resampled : ndarray, shape (n_samples_new, n_features)
+            The array containing the resampled data.
+
+        y_resampled : ndarray, shape (n_samples_new)
+            The corresponding label of `X_resampled`
+
+        idx_under : ndarray, shape (n_samples, )
+            If `return_indices` is `True`, a boolean array will be returned
+            containing the which samples have been selected.
+
+        """
+        # Check the consistency of X and y
+	X, y = check_X_y(X, y)
```

Comment:
weird, here it seems normal ... I believe autopep8 will handle this.

![screenshot from 2016-05-30 16 07 48](https://cloud.githubusercontent.com/assets/511783/15656741/af0f119c-2680-11e6-8a5b-065845d57b7c.png)